### PR TITLE
4.2.1: Build for grpc-1.43

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ requirements:
     - {{ compiler('cxx') }}
     - sed               # [unix]
     - openjdk >=8       # [unix]
-    - grpc-cpp          # [unix]
+    - grpc-cpp 1.43         # [unix]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:33]: I thought you wanted a migrator?

Comment (xhochy) [reply recipe/meta.yaml:33]: No, I wanted a migrator for (C++14 abseil + grpc-cpp 1.45, C++17 abseil + grpc-cpp 1.46). 

The build here is to unblock the `libprotobuf=3.20` migration without touching the current `grpc-cpp` pin. After this PR is merged, I would revert the explicit grpc-cpp pin again and do the above mentioned migrator.

Comment (hmaarrfk) [reply recipe/meta.yaml:33]: understood.

Comment (xhochy) [reply recipe/meta.yaml:33]: Yes!

**Pull request merged**